src/ucs/arch/aarch64/cpu.h

```diff
@@ -231,10 +235,30 @@ static inline void ucs_arch_clear_cache(void *start, void *end)
 }
 #endif
 
+#if defined(__ARM_FEATURE_SVE)
+static inline void *memcpy_aarch64_sve(void *dest, const void *src, size_t len)
+{
+    uint8_t *dest_u8      = (uint8_t*) dest;
+    const uint8_t *src_u8 = (uint8_t*) src;
+    uint64_t i            = 0;
+    svbool_t pg           = svwhilelt_b8_u64(i, (uint64_t)len);
+
+    do {
```

Comment:
does this memcpy support len==0?

Reply:
Yes, because `whilelt` returns a predicate of all zeros when `i >= len`, which means in the first and only iteration, all lanes are inactive. According to [ARM doc](https://developer.arm.com/docs/ddi0596/h/a64-sve-instructions-alphabetic-order/st1b-scalar-plus-scalar-contiguous-store-bytes-from-vector-scalar-index), the memory is not touched.

I have also tested this implementation with zero copy length, using random `dest` and `src`, and the destination array always remain unchanged.